### DISCUSSION TITLE
Extend Soil Model

### DISCUSSION THREAD

**Discussion by JayfansChou** [Anura3D/Anura3D_OpenSource#246] [Q&A]
Dear all,

I encountered the following error prompt when compiling the user-defined UMAT subroutine from a .for file to a .dll file. I suspect this is because I introduced Utility Routines from ABAQUS, such as SPRINC and SPRIND, into the UMAT, but didn’t provide their source code, leading to the error. How can I solve this problem?
![error_message](https://github.com/user-attachments/assets/ff8e241f-982e-4b45-b45a-9b471847f208)

Best Regards, Chou

**Comment by zorroxt1118** [comment]
This document might be helpful:
https://imechanica.org/files/Writing%20a%20UMAT.pdf

**Reply by JayfansChou** [reply]
Thank you very much. I will have a look.
